Source/WebCore/platform/graphics/FontCascade.cpp

```diff
@@ -330,7 +330,9 @@ GlyphData FontCascade::glyphDataForCharacter(UChar32 c, bool mirror, FontVariant
     if (mirror)
         c = u_charMirror(c);
 
-    return m_fonts->glyphDataForCharacter(c, m_fontDescription, variant);
+    auto emojiPolicy = resolveEmojiPolicy(FontVariantEmoji::Normal, c);
```

Comment:
Is this going to be fast enough to do with no caching?

Reply:
Well, Core Text does this without any caching, and it's not super obvious if a hash lookup is significantly faster than a property lookup, soooooo probably yes